src/components/AccountDropdown/AccountDropdown.tsx

```diff
@@ -246,6 +246,7 @@ export const AccountDropdown: FC<AccountDropdownProps> = memo(
         validatedAccountIdFromArgs ??
         (autoSelectHighestBalance ? highestUserCurrencyBalanceAccountId : undefined) ??
         firstAccountId
+      if (previousSelectedAccountId === preSelectedAccountId) return
```

Comment:
The second guts of this against infinite loops, the only reason why this was working previously was because we accidentally were not reactive on accountIds in this component.
Failure to do so would produce an infinite loop on account selection (tested in RFOX, but pretty sure it would across the app): https://jam.dev/c/e64b8853-61a0-4c01-a1a5-1363885ac7d3